Adds Button component and supports button "type" attribute

abobwhite opened the pull request:
Let me know if you'd like me to handle this differently but I merged a branch of another fork which created a Button component to reuse everywhere <button /> was used. I then slightly enhanced that by adding a new argument that Button can take: "type". The type attribute is important because the default value of the attribute within forms is "submit" which is almost always not what you want. The component defaults "type" to "button" if not defined

gabrielrtakeda commented on src/package/SlateEditor/SlateEditor.js:
```diff
@@ -15,7 +15,11 @@ class SlateEditor extends Component {
-    this.setState({ state })
+    const { onChange } = this.props;
+    this.setState({ state });
+    if(onChange) {
```
To call the `onChange` like a function below, may it would be better to check if the `onChange` prop is really a prop of function type... What do you think to do something like:

```js
onChange && onChange.constructor === Function && onChange(state)
```

And, if you can remove the semicolons it will be great too (:

abobwhite replied:
@gabrielrtakeda Why no semi-colons? I write a lot of JavaScript and I've never heard of semi-colons not being used...?

gabrielrtakeda replied:
@abobwhite Just a preference and a convention that we have adopted on the project.. Nothing in particular with the JS..
https://medium.com/@kentcdodds/semicolons-in-javascript-a-preference-dd8fc8b80895

gabrielrtakeda commented on src/package/components/button/Button.js:
```diff
@@ -0,0 +1,23 @@
+    <button
+      id={id}
+      style={style}
+      onClick={preventDefault(onClick)}
```
Instead of creating a function to pass a **callback**, may it will be better to check it directly on the prop?
What do you think something like:

```js
<button
  onClick={e => {
    onClick && onClick.constructor === Function && onClick(e) // <~ pass the event object
  }}
/>
```

And, instead of using directly the `e.preventDefault` on the componente, as we passing the event object to the `onClick` function, may would be better to prevent it on the function that we are passing to the component, what do you think?

Something like:

```jsx
// using the Button component, e.preventDefault in onClick prop
import { Button } from '../components/button'

const ButtonExample = (props) => (
  <Button
    {...props}
    onClick={e => {
      e.preventDefault()
      //
      // Do what you need to do
      // 
    }}
  />
)
```

gabrielrtakeda commented on src/package/components/button/Button.js:
```diff
@@ -0,0 +1,23 @@
+};
+
+export default ({ children, id, onClick, className, style, type }) => {
+  const buttonType = type || 'button'; // Make button default so that when used in forms, "submit" does not become default
```
Instead of creating a variable inside the component, it may would be better to define the expected default value directly to the component props like:

```jsx
const Button = ({ ...otherProps, type }) => (
  // 
  // Button component implementation
  // 
)

// here we define the expected default value to the `type` prop
Button.defaultProps = { 
  type: 'button'
}

export default Button
```

gabrielrtakeda commented:
@abobwhite Can you remove the changes made by the build command **lib/** dir, please? This process will be automatically made when we release a new version to npm...

gabrielrtakeda commented on src/package/plugins/slate-alignment-plugin/AlignmentCenterButton.js:
```diff
@@ -2,11 +2,12 @@ import React from 'react'
 
 const AlignmentCenterButton = ({ state, onChange, changeState, className, style }) => (
-  <button
+  <Button
```
@abobwhite Can you add the `type` prop to all the plugins button to provide a way to change the type of button component, since now we can edit the button's type?

abobwhite commented:
@gabrielrtakeda The PR is updated. I also squashed and renamed my commits but it appears that the PR is still showing those original commits. Surely when you merge, you can rebase, though
